src/upload/form/mod.rs

```diff
@@ -70,23 +75,75 @@ pub fn build(
 
             service_worker::build_form(&assets, session_config)
         }
-        TargetType::JavaScript => {
-            log::info!("JavaScript project detected. Publishing...");
-            let package_dir = target.package_dir()?;
-            let package = Package::new(&package_dir)?;
-
-            let script_path = package.main(&package_dir)?;
-
-            let assets = ServiceWorkerAssets::new(
-                script_path,
-                wasm_modules,
-                kv_namespaces.to_vec(),
-                text_blobs,
-                plain_texts,
-            )?;
-
-            service_worker::build_form(&assets, session_config)
-        }
+        TargetType::JavaScript => match &target.build {
+            Some(config) => match &config.upload_format {
+                ScriptFormat::ServiceWorker => {
+                    log::info!("Plain JavaScript project detected. Publishing...");
+                    let package_dir = target.package_dir()?;
+                    let package = Package::new(&package_dir)?;
+                    let script_path = package.main(&package_dir)?;
+
+                    let assets = ServiceWorkerAssets::new(
+                        script_path,
+                        wasm_modules,
+                        kv_namespaces.to_vec(),
+                        text_blobs,
+                        plain_texts,
+                    )?;
+
+                    service_worker::build_form(&assets, session_config)
+                }
+                ScriptFormat::Modules => {
+                    let package_dir = target.package_dir()?;
+                    let package = Package::new(&package_dir)?;
+                    let main_module = package.module(&package_dir)?;
+                    let main_module_name = filename_from_path(&main_module)
+                        .ok_or_else(|| failure::err_msg("filename required for main module"))?;
+
+                    let modules_iter = WalkBuilder::new(config.upload_dir.clone())
+                        .standard_filters(false)
+                        .ignore(true)
```

Comment:
Added support for a `.ignore` file in the upload directory, and ignored hidden files by default (which takes care of making sure `.ignore` doesn't get uploaded).

I don't feel strongly about having this here vs a section in `wrangler.toml`, but looking at `.ignore` was awfully convenient to add :p

Reply:
Removed the .ignore file support in favor of a section in `wrangler.toml`, since getting `.ignore` in version control without getting the rest of the upload directory in is kinda an annoying thing to deal with.